syft/frameworks/torch/mpc/spdz.py

```diff
@@ -1,57 +1,164 @@
-from typing import Callable
-
-import torch
+import asyncio
+import math
+import multiprocessing
+import torch as th
 
 import syft as sy
-from syft.frameworks.torch.mpc.beaver import request_triple
-from syft.workers.abstract import AbstractWorker
+from syft.exceptions import EmptyCryptoPrimitiveStoreError
+from syft.generic.utils import allow_command
+from syft.generic.utils import remote
+
+from syft.frameworks.torch.mpc.fss import N_CORES
 
 no_wrap = {"no_wrap": True}
 
 
-def spdz_mul(cmd: Callable, x_sh, y_sh, crypto_provider: AbstractWorker, field: int, dtype: str):
-    """Abstractly multiplies two tensors (mul or matmul)
+def full_name(f):
+    return f"syft.frameworks.torch.mpc.spdz.{f.__name__}"
+
+
+# share level
+@allow_command
+def spdz_mask(x, y, op, dtype, torch_dtype, field):
+    a, b, c = x.owner.crypto_store.get_keys(
+        op=op,
+        shapes=(x.shape, y.shape),
+        n_instances=1,
+        remove=False,
+        dtype=dtype,
+        torch_dtype=torch_dtype,
+        field=field,
+    )
+    return x - a, y - b
+
+
+def slice(x, j, slice_size):
+    x_slice = x[j * slice_size : (j + 1) * slice_size]
+    x_slice.owner = x.owner
+    return x_slice
+
+
+def triple_mat_mul(core_id, delta, epsilon, a, b):
+    cmd = th.matmul
+    delta_b = cmd(delta, b)
+    a_epsilon = cmd(a, epsilon)
+    delta_epsilon = cmd(delta, epsilon)
+    return core_id, delta_b, a_epsilon, delta_epsilon
+
+
+# share level
+@allow_command
+def spdz_compute(j, delta, epsilon, op, dtype, torch_dtype, field):
```

Comment:
Q: Could you add a description? Not sure what is ```j```

Reply:
Sure!